client/my-sites/site-settings/form-general.jsx

```diff
@@ -533,6 +533,84 @@ export class SiteSettingsFormGeneral extends Component {
 		);
 	}
 
+	handleSupportAccessChange = ( { isAllowed } ) => {
+		const SUPPORT_ACCESS_EXPIRY_IN_DAYS = 30;
+		const { updateFields } = this.props;
+		if ( isAllowed ) {
+			const today = new Date();
+			const expiry =
+				( today.setDate( today.getDate() + SUPPORT_ACCESS_EXPIRY_IN_DAYS ) / 1000 ) | 0;
+			updateFields( { support_access_allowed_until: expiry, allow_support_access: 1 } );
+		} else {
+			updateFields( { support_access_allowed_until: '', allow_support_access: 0 } );
+		}
+	};
+
+	supportAccessOptions() {
+		const { fields, isRequestingSettings, translate } = this.props;
+		const isSupportAccessAllowed = fields.allow_support_access === 1;
+
+		return (
+			<FormFieldset>
+				<FormLabel>
+					<FormRadio
+						name="allow_support_access"
+						value="1"
+						checked={ isSupportAccessAllowed }
+						onChange={ () =>
+							this.handleSupportAccessChange( {
+								isAllowed: true,
+							} )
+						}
+						disabled={ isRequestingSettings }
+						label={ translate( 'Allow us to access your P2 to provide support' ) }
+					/>
+				</FormLabel>
+				<FormSettingExplanation isIndented>
+					{ translate( 'This will deactivate automatically in 30 days.' ) }
+				</FormSettingExplanation>
+
+				<FormLabel>
+					<FormRadio
+						name="allow_support_access"
+						value="0"
+						checked={ ! isSupportAccessAllowed }
+						onChange={ () =>
+							this.handleSupportAccessChange( {
+								isAllowed: false,
+							} )
+						}
+						disabled={ isRequestingSettings }
+						label={ translate( "Don't allow (recommended)" ) }
+					/>
+				</FormLabel>
+				<FormSettingExplanation isIndented>
+					{ translate( "We won't be able to access your P2 if you reach out for help." ) }
+				</FormSettingExplanation>
+			</FormFieldset>
+		);
+	}
+
+	supportAccessSettings() {
+		const { isRequestingSettings, translate, handleSubmitForm, isSavingSettings } = this.props;
+
+		return (
+			<>
+				<SettingsSectionHeader
+					disabled={ isRequestingSettings || isSavingSettings }
+					id="support-access-settings"
+					isSaving={ isSavingSettings }
+					onButtonClick={ handleSubmitForm }
+					showButton
+					title={ translate( 'Support Access', { context: 'Support Access settings header' } ) }
```

Comment:
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?